ibm/config.go

```diff
@@ -1109,11 +1109,23 @@ func (c *Config) ClientSession() (interface{}, error) {
 	if c.Visibility == "private" || c.Visibility == "public-and-private" {
 		kpurl = contructEndpoint(fmt.Sprintf("private.%s.kms", c.Region), cloudEndpoint)
 	}
-	options := kp.ClientConfig{
-		BaseURL:       envFallBack([]string{"IBMCLOUD_KP_API_ENDPOINT"}, kpurl),
-		Authorization: sess.BluemixSession.Config.IAMAccessToken,
-		// InstanceID:    "42fET57nnadurKXzXAedFLOhGqETfIGYxOmQXkFgkJV9",
-		Verbose: kp.VerboseFailOnly,
+	var options kp.ClientConfig
+	config := sess.BluemixSession.Config
```

Comment:
instead of  this variable config := sess.BluemixSession.Config ,... you can use c.BluemixAPIKey != ""